dask_ml/model_selection/_incremental.py

```diff
@@ -198,34 +198,40 @@ async def _fit(
     else:
         y_test = await client.scatter(y_test)
 
-    # Convert to batches of delayed objects of numpy arrays
-    X_train = sorted(futures_of(X_train), key=lambda f: f.key)
-    y_train = sorted(futures_of(y_train), key=lambda f: f.key)
-    assert len(X_train) == len(y_train)
-
-    train_eg = await client.gather(client.map(len, y_train))
-    msg = "[CV%s] For training there are between %d and %d examples in each chunk"
-    logger.info(msg, prefix, min(train_eg), max(train_eg))
+    if hasattr(X_train, 'npartitions'):
+        # Convert to batches of delayed objects of numpy arrays
+        X_train = sorted(futures_of(X_train), key=lambda f: f.key)
+        y_train = sorted(futures_of(y_train), key=lambda f: f.key)
+        assert len(X_train) == len(y_train)
+
+        train_eg = await client.gather(client.map(len, y_train))
+        msg = "[CV%s] For training there are between %d and %d examples in each chunk"
+        logger.info(msg, prefix, min(train_eg), max(train_eg))
```

Comment:
Why won't this code work?

``` python
train_eg =  ...
msg = ...
if len(train_eg):
    logger.info(msg, prefix, min(train_eg), max(train_eg))
```

This avoids `min([])` as mentioned in https://github.com/dask/dask-ml/issues/748#issuecomment-717814887.

Then, I think the if-statement could be moved inside `get_futures`:

``` python
def get_futures(partial_fit_calls):
    if not isinstance(X_train, da.Array):
        return X_train, y_train
    ...  # existing implementation
```

Reply:
you are right, I was not careful when I red your comment, this would work.

The only reason why I did not implement it that way is because saying `if len(train_eg):` or even shorter `if train_eg` is the same as checking `isinstance(X_train, da.Array)` because `train_eg` will be empty only if `X_train` is not a `da.Array`. This means that in your implementation you are still executing these lines, even if not needed:

```
X_train = sorted(futures_of(X_train), key=lambda f: f.key)
y_train = sorted(futures_of(y_train), key=lambda f: f.key)
assert len(X_train) == len(y_train)

train_eg = await client.gather(client.map(len, y_train))
```

Reply:
I think the logging message should be issued if the dataset object has an implementation of `__len__`. The computation of `train_eg` only happens once (`train_eg` is a list of ints).

I don't like two separate definitions of `get_futures`, especially if the both return training data.